bundles/org.openhab.binding.modbus.studer/src/main/java/org/openhab/binding/modbus/studer/internal/StuderHandler.java

```diff
@@ -0,0 +1,432 @@
+            return null;
+        }
+
+        ThingHandler handler = bridge.getHandler();
+        if (handler == null) {
+            logger.debug("Bridge handler is null");
+            return null;
+        }
+
+        if (handler instanceof ModbusEndpointThingHandler) {
+            ModbusEndpointThingHandler slaveEndpoint = (ModbusEndpointThingHandler) handler;
+            return slaveEndpoint;
+        } else {
+            logger.debug("Unexpected bridge handler: {}", handler);
+            return null;
+        }
+    }
+
+    /**
+     * Get a reference to the modbus endpoint
+     */
+    private void connectEndpoint() {
+        if (comms != null) {
+            return;
+        }
+
+        ModbusEndpointThingHandler slaveEndpointThingHandler = getEndpointThingHandler();
+        if (slaveEndpointThingHandler == null) {
+            @SuppressWarnings("null")
+            String label = Optional.ofNullable(getBridge()).map(b -> b.getLabel()).orElse("<null>");
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE,
+                    String.format("Bridge '%s' is offline", label));
+            logger.debug("No bridge handler available -- aborting init for {}", label);
+            return;
+        }
+        comms = slaveEndpointThingHandler.getCommunicationInterface();
+        if (comms == null) {
+            @SuppressWarnings("null")
+            String label = Optional.ofNullable(getBridge()).map(b -> b.getLabel()).orElse("<null>");
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE,
+                    String.format("Bridge '%s' not completely initialized", label));
+            logger.debug("Bridge not initialized fully (no endpoint) -- aborting init for {}", this);
+            return;
+        }
+    }
+
+    /**
+     * Remove the endpoint if exists
+     */
+    private void unregisterEndpoint() {
+        // Comms will be close()'d by endpoint thing handler
+        comms = null;
+    }
+
+    private synchronized void unregisterPollTasks() {
+        if (pollTasks.isEmpty()) {
+            return;
+        }
+        logger.debug("Unregistering polling from ModbusManager");
+        @Nullable
+        ModbusCommunicationInterface mycomms = comms;
+        if (mycomms != null) {
+            for (PollTask t : pollTasks) {
+                mycomms.unregisterRegularPoll(t);
+            }
+            pollTasks.clear();
+        }
+    }
+
+    /**
+     * Register poll task
+     * This is where we set up our regular poller
+     */
+    @SuppressWarnings("null")
+    private synchronized void registerPollTask(int registerNumber) {
+        if (pollTasks.size() >= registers.length) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR);
+            throw new IllegalStateException("New pollTask invalid");
+        }
+        @Nullable
+        ModbusCommunicationInterface mycomms = comms;
+        @Nullable
```

Comment:
These annotations can be removed. The compiler can detect it on local variables itself.